cyral/resource.go

```diff
@@ -74,11 +75,22 @@ func ReadResource(config ResourceOperationConfig) schema.ReadContextFunc {
 
 		body, err := c.DoRequest(url, config.HttpMethod, nil)
 		if err != nil {
+
+			if config.UnsetIdOnError {
+				d.SetId("")
+				return nil
+			}
```

Comment:
Could you please elaborate on how this will prevent the issue described in the PR title? I thought the only change required is how we handle the ID.

Reply:
If we don't do this, everything breakes once something is deleted upstream, and it's a simple change, backwards compatible that allows for lot's of little future improvements in the behavior for our resources.